runtime/protocol/http/common/src/aws/smithy/kotlin/runtime/http/auth/HttpSigner.kt

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package aws.smithy.kotlin.runtime.http.auth
+
+import aws.smithy.kotlin.runtime.client.ExecutionContext
+import aws.smithy.kotlin.runtime.http.request.HttpRequestBuilder
+import aws.smithy.kotlin.runtime.util.InternalApi
+
+/**
+ * Represents a component capable of signing an HTTP request
+ */
+@InternalApi
+public interface HttpSigner {
+    public companion object {
+        /**
+         * A no-op signer that does nothing with the request
+         */
+        public val NONE: HttpSigner = object : HttpSigner {
```

Comment:
nit: `NONE` or `None`? would `NoOp` be a clearer name?

if this is just temporary, please ignore

Reply:
renamed to `Anonymous` which aligns with the ref arch terminology